src/integration/integration_scheduler_scalingengine_test.go

```diff
@@ -0,0 +1,161 @@
+
+		startFakeCCNOAAUAA(initInstanceCount)
+
+		scalingEngineConfPath = components.PrepareScalingEngineConfig(dbUrl, components.Ports[ScalingEngine], fakeCCNOAAUAA.URL(), cf.GrantTypePassword, tmpDir)
+		schedulerConfPath = components.PrepareSchedulerConfig(dbUrl, fmt.Sprintf("https://127.0.0.1:%d", components.Ports[ScalingEngine]), tmpDir)
+		schedulerProcess = startScheduler()
+		startScalingEngine()
+
+		policyByte := readPolicyFromFile("fakePolicyWithSpecificDateSchedule.json")
+		policyStr = setPolicyDateTime(policyByte)
+
+	})
+
+	AfterEach(func() {
+		deleteSchedules(testAppId)
+		stopAll()
+		stopScheduler(schedulerProcess)
+	})
+
+	Describe("Create Schedule", func() {
+		Context("Valid specific date schedule", func() {
+
+			It("creates active schedule in scaling engine", func() {
+				Expect(getNumberOfActiveSchedules(testAppId)).To(Equal(0))
+				resp, err := createSchedule(testAppId, policyStr)
+				Expect(err).NotTo(HaveOccurred())
+				defer resp.Body.Close()
+
+				body, err := ioutil.ReadAll(resp.Body)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(string(body)).To(Equal(""))
+				Expect(resp.StatusCode).To(Equal(http.StatusOK))
+				Eventually(getAppIdOfActiveSchedules, 5*time.Minute).Should(ContainElement(testAppId))
+				Expect(getNumberOfActiveSchedules(testAppId)).To(Equal(1))
+
+			})
+		})
+
+		Context("ScalingEngine Server is down", func() {
+			BeforeEach(func() {
+				stopAll()
+			})
+
+			It("should return 500", func() {
+				resp, err := createSchedule(testAppId, policyStr)
+				Expect(err).NotTo(HaveOccurred())
+				defer resp.Body.Close()
+
+				body, err := ioutil.ReadAll(resp.Body)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(string(body)).To(Equal(""))
+				Expect(resp.StatusCode).To(Equal(http.StatusOK))
+				Eventually(stdOutbufferMap[Scheduler], 5*time.Minute).Should(gbytes.Say("Error connecting to scaling engine, failed with error: .* for app id: " + testAppId))
+				Expect(getNumberOfActiveSchedules(testAppId)).To(Equal(0))
+			})
+		})
+
+	})
+
+	Describe("Delete Schedule", func() {
+		BeforeEach(func() {
+			resp, err := createSchedule(testAppId, policyStr)
+			Expect(err).NotTo(HaveOccurred())
+			defer resp.Body.Close()
+
+			body, err := ioutil.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(body)).To(Equal(""))
+			Expect(resp.StatusCode).To(Equal(http.StatusOK))
+			Eventually(getAppIdOfActiveSchedules, 5*time.Minute).Should(ContainElement(testAppId))
+		})
+
+		It("deletes active schedule in scaling engine", func() {
+			Expect(getNumberOfActiveSchedules(testAppId)).To(Equal(1))
+			resp, err := deleteSchedule(testAppId)
+			Expect(err).NotTo(HaveOccurred())
+			defer resp.Body.Close()
+
+			body, err := ioutil.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(body)).To(Equal(""))
+			Expect(resp.StatusCode).To(Equal(http.StatusNoContent))
+			Eventually(getAppIdOfActiveSchedules, 5*time.Minute).ShouldNot(ContainElement(testAppId))
```

Comment:
Why not just combine the 2 lines.
Eventually(func() int {
return getNumberOfActiveSchedules...
}).Should(BeZero())